### PR TITLE
Remove cheerio dependency

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~Added/Updated related documentation.~
- [X] Added/Updated related tests.

## Short description of the change(s)

Ref #1499 

Part of the work to reduce bundle size for the browser extension.
Required fixing an issue with recognized types for JavaScript. 

~Also enabled a new test in `hint-meta-charset-utf-8` for reporting
unnecessary XML declarations (previously a TODO).~
(abandoned due to mime-type detection issues)

### CODE REVIEW

File: packages/hint-meta-charset-utf-8/package.json [hunk 1]
```diff
@@ -6,13 +6,10 @@
     ],
     "timeout": "1m"
   },
-  "dependencies": {
-    "cheerio": "^1.0.0-rc.2"
-  },
   "description": "hint for best practices related to the meta charset",
   "devDependencies": {
+    "@hint/parser-html": "^2.0.2",
```
Comment (molant) [inline packages/hint-meta-charset-utf-8/package.json]: Should we just add this dependency to `utils-tests-helpers` and add the `html` parser directly to the configurations?

**Comment by molant** [comment]
>   ✖ [jsdom] The XML charset declaration is used' No results found, should be 1

👆 this test is failing

**Comment by antross** [comment]
> > ✖ [jsdom] The XML charset declaration is used' No results found, should be 1
> 
> 👆 this test is failing

It's because the test doesn't specify a mime-type and it gets inferred as `text/xml`. This ties back to the mime-type inconsistencies we were talking about between connectors earlier today. 

I'm going to fix this by explicitly specifying `text/html` as the mime-type in the test.

**Comment by molant** [comment]
> I'm going to fix this by explicitly specifying `text/html` as the mime-type in the test.

How much sense does it make taking into consideration that the test is `The XML charset declaration is used`? Maybe we should just skip it for now?

File: packages/hint-meta-charset-utf-8/package.json (hunk 1 above)
Comment (antross) [reply packages/hint-meta-charset-utf-8/package.json]: In my other PR we added `html` directly to any configuration passed to `utils-tests-helpers`.

That said, I'll still need `parser-html` as a `devDependency` here in order to get the correct types when listening for the `parse::end::html` event. I just don't have to explicitly specify it in the tests.

Comment (molant) [reply packages/hint-meta-charset-utf-8/package.json]: Got it.

**Comment by antross** [comment]
> How much sense does it make taking into consideration that the test is `The XML charset declaration is used`? Maybe we should just skip it for now?

I think it makes sense since the problem is when someone specifies the XML declaration in HTML content. If it were an actual XML document the declaration would be perfectly fine.

**Comment by antross** [comment]
> I think it makes sense since the problem is when someone specifies the XML declaration in HTML content. If it were an actual XML document the declaration would be perfectly fine.

Until I discovered an XML declaration causes us to treat content as XML internally regardless of mime-type (this differs from browser behavior). I'll add a note to #2067 and drop this for now.

**Pull request merged**